### PR TITLE
Refactor VMs per template card to be namespaced

### PR DESCRIPTION
**Pull request by pcbailey**
## 📝 Description

This PR refactors the VMs per template card to display data based on the selected namespace for admins and non-admins or all namespaces for admins.

### CODE REVIEW

**Comment by pcbailey** [comment]
/hold

Holding until https://github.com/openshift/console/pull/12001 passes and the namespace bar can be added to the overview page.

**Comment by pcbailey** [comment]
@avivtur @hstastna @metalice @upalatucci @vojtechszocs PTAL

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      (acc, key) => {
+        acc[key] = resourceData[key]?.data;
+        return acc;
+      },
+      { loaded: false, vms: [], templates: [] },
+    );
+
+    resources.loaded = Object.values(resourceData).every((resource) => resource?.loaded);
+    return resources;
+  }, [resourceData]);
```
[the remark this answers is not in the document]
Comment (pcbailey) [reply src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts]: Good point. Fixed!

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx [hunk 2]
```diff
@@ -15,17 +17,22 @@ import {
...
-import './RunningVMsPerTemplateCard.scss';
+import './VMsPerTemplateCard.scss';
 
-const RunningVMsPerTemplateCard = () => {
+const VMsPerTemplateCard = () => {
   const { t } = useKubevirtTranslation();
-  const useRunningVMsPerTemplatesResources = useRunningVMsPerTemplatesHook();
-  const { loaded, vms, templates } = useRunningVMsPerTemplatesResources();
+  const [activeNamespace] = useActiveNamespace();
+  const namespace = React.useMemo(
```
Comment (metalice) [inline src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx:30]: u removed the astrix from react import but imported Rect. :)

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  const resourceData = useK8sWatchResources<{ [key: string]: K8sResourceCommon[] }>(
+    watchedResources,
+  );
+
+  return useMemo(() => {
+    return {
+      templates: resourceData.templates.data,
+      vms: resourceData.vms.data,
+      loaded: resourceData.templates.loaded && resourceData.vms.loaded,
```
Comment (metalice) [inline src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts:49]: please add ?. , im not sure if those nested props are guaranteed using watchResourceS

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx [hunk 4]
```diff
@@ -15,17 +17,22 @@ import {
...
+    () => (activeNamespace === ALL_NAMESPACES_SESSION_KEY ? null : activeNamespace),
+    [activeNamespace],
+  );
+  const { loaded, vms, templates } = useVMsPerTemplateResources(namespace);
```
Comment (metalice) [inline src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx:34]: maybe useVMsPerTemplateResources can use the active namespace hook? I don't see anywhere else in this file namespace is used, but maybe I'm missing :)

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts (hunk 3 above)
Comment (pcbailey) [reply src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts]: Done!

File: src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx (hunk 4 above)
Comment (pcbailey) [reply src/views/clusteroverview/OverviewTab/running-vms-per-template-card/VMsPerTemplateCard.tsx]: Once again sir, you are correct. =) Fixed!

**Pull request merged**